analyses/building-trajectories-monocle/01-building-trajectories-monocle.Rmd

```diff
@@ -0,0 +1,299 @@
+
+# Set up
+
+```{r load-library, echo=TRUE}
+attach(params)
+suppressPackageStartupMessages({
+  library(Signac)
+  library(Seurat)
+  library(SeuratWrappers)
+  library(monocle3)
+  library(Matrix)
+  library(tidyverse)
+  library(patchwork)
+  library(base)
+  library(knitr)
+}) 
+```
+
+# Directories and paths to file Inputs/Outputs
+
+```{r set-dir-and-file-names, echo=TRUE}
+module_analysis_dir <- file.path(root_dir, "analyses", "building-trajectories-monocle") 
+analysis_dir <- file.path(root_dir, "analyses", "integration-with-scrna-seq-data") 
+data_dir <- file.path(analysis_dir, "results", "01_integration_with_scrna_seq_data")
+
+# Input files
+data_file <- file.path(data_dir, "seurat_obj_reference.rds")
+
+# Create plots directory
+module_plots_dir <- file.path(module_analysis_dir , "plots")
+
+plots_dir <- file.path(module_plots_dir, "01_building_trajectories_monocle") 
+if (!dir.exists(plots_dir)) {
+  dir.create(plots_dir)}
+
+# Source functions
+source(paste0(module_analysis_dir, "/util/monocle3_updated.R"))
+```
+
+```{r echo=FALSE, warning=FALSE}
+opts_chunk$set(fig.align='center',
+               external=TRUE,
+               echo=FALSE,
+               warning=FALSE,
+               fig.pos='H')
+a4width <- 8.3
+a4height <- 11.7
+```
+
+# Read seurat object
+
+First, we will use the seurat object as generated from the pipeline in the `integration-with-scrna-seq-data` module. 
+
+```{r read-object, fig.width = 6, fig.height = 5, fig.fullwidth = TRUE, echo=TRUE}
+seurat_obj <- readRDS(data_file)
+DefaultAssay(seurat_obj) <- assay
+
+name <- paste0(plots_dir, "/", glue::glue("DimPlot-clusters.png"), sep = "")
+print(DimPlot(seurat_obj, label = TRUE) + NoLegend())
+ggsave(file = name, width = 6, height = 5, device = "png")
+```
+
+Assign each cluster to the most common cell type based on the previous predicted annotations from the `./integration-with-scrna-seq-data/01-integration-with-scrna-seq-data.Rmd` step.
+
+```{r assign-celltypes, fig.width = 6, fig.height = 5, fig.fullwidth = TRUE, echo=TRUE}
+for(i in levels(seurat_obj)) {
+  cells_to_reid <- WhichCells(seurat_obj, idents = i)
+  newid <- names(sort(table(seurat_obj$predicted.id[cells_to_reid]),decreasing=TRUE))[1]
+  Idents(seurat_obj, cells = cells_to_reid) <- newid
+}
+seurat_obj$assigned_celltype <- Idents(seurat_obj)
+
+name <- paste0(plots_dir, "/", glue::glue("DimPlot-assigned-celltype.png"), sep = "")
+print(DimPlot(seurat_obj, label = TRUE))
+ggsave(file = name, width = 6, height = 5, device = "png")
+```
+
+Next we can subset the different lineages and create a trajectory for each lineage. Another way to build the trajectories is to use the whole dataset and build separate pseudotime trajectories for the different cell partitions found by Monocle 3.
+
+```{r subset-celltypes, echo=TRUE}
+lineage1 <- seurat_obj[,  seurat_obj$assigned_celltype %in% lineage1_value_vector]
+lineage2 <- seurat_obj[, seurat_obj$assigned_celltype %in% lineage2_value_vector]
+```
```

Comment:
It might be worth considering allowing more lineages for future updates, if it would be useful.